datafusion/common/src/column.rs

```diff
@@ -156,6 +157,17 @@ impl Column {
         }
     }
 
+    fn quoted_flat_name_if_contain_dot(&self) -> String {
+        match &self.relation {
+            Some(r) => format!(
+                "{}.{}",
+                table_reference_to_quoted_string(r),
+                quoted_if_contain_dot(&self.name)
+            ),
+            None => quoted_if_contain_dot(&self.name).to_string(),
+        }
+    }
```

Comment:
@jayzhan211  Before I fix other tests, I want to check if this behavior makes sense. (It involves too many tests 😢 ).
Now, we only quote an identifier if it contains the dot. However, some cases like `sum(t1.c1)` will also be quoted, even if it's a function call. I think it's not worth doing more checking to exclude this kind of case. What do you think?

Reply:
I think it is not ideal if `sum(t1.c1)` is quoted. I hope the change is minimum and not effect others.

Reply:
Maybe instead of modifying `Column`, we should modify the `display_name` for `Expr`, so if we found column inside ScalarFunction, we could skip the double quote anyway. (by something like boolean flag?)

Reply:
> Maybe instead of modifying `Column`, we should modify the `display_name` for `Expr`, so if we found column inside ScalarFunction, we could skip the double quote anyway. (by something like boolean flag?)

I'm not sure if it's that simple 🤔. In my experience, the column might look like this:
```
Column { qualifier: None, name: "sum(t1.c1)" }
```
I think it's hard to find a consistent pattern for it because we use many `Column::from_name` calls to create projections. For example, in https://github.com/apache/datafusion/blob/f4e519f9df9ab5972638d3f2743da01887a52668/datafusion/sql/src/relation/join.rs#L118
the column name could be complex and unruly.